datalad/runner/nonasyncrunner.py

```diff
@@ -291,6 +303,13 @@ def run(self) -> Union[Any, Generator]:
                # get the return code of the process
                result = gen.return_code
         """
+        with self.initialization_lock:
+            return self._locked_run()
```

Comment:
Fine with me but why not that alternative of @fasteners.locked?

Reply:
I guess it is a matter of taste. I find the @fastener.locked-approach with the self._lock attribute less transparent

Reply:
ok. May be some time we will come with a design doc to unify the tastes on this regard ;)